src/internet_identity_interface/src/http_gateway.rs

```diff
@@ -1,17 +1,22 @@
 //! Types as defined by the HTTP gateway spec.
 //! See https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-gateway-interface
 
-use candid::{CandidType, Deserialize, Func};
+use candid::{define_function, CandidType, Deserialize};
 use serde_bytes::ByteBuf;
 
 pub type HeaderField = (String, String);
 
 #[derive(Clone, Debug, CandidType, Deserialize)]
 pub struct Token {}
 
+define_function!(pub StreamingCallbackFunction : (Token) -> (StreamingCallbackHttpResponse) query);
```

Comment:
Can you explain this change?

Reply:
Previously, the typings for the streaming callback function were inaccurate (because we just had the raw `Func` type because it was super cumbersome to actually define it properly). But it didn't matter, since we don't use the full set of all HTTP gateway interactions.

Now, the languages team made this (common) shortcut error out at runtime and instead provides better functionality (namely the `define_function!` macro) to conveniently type a function including the arguments, return values and the mode (i.e. query).

So essentially cleans up an old piece of tech debt.